droid-hal-device.inc

```diff
@@ -538,6 +538,7 @@ echo "" > tmp/droid-hal.files
 fi
 
 if [ -f %{android_root}/out/target/product/%{device}/system/etc/init/hybris_extras.rc ]; then
```

Comment:
If `.../etc/init` should never exist in Android 6 (or earlier) adaptations, why not move this part into the if block just above and re-use the mkdir? ^

Reply:
As I worked through HADK and built hybris-hal for hammerhead (CyanogenMod 11.0 / Android 4.4). ./out/target/product/hammerhead/system/etc/init/hybris_extras.rc was created.

Reply:
@krnlyng, @mlehtima, could you please tell us if there is any part in hybris Android 6 (and earlier) adaptations which would read and/or run the .rc files under `/usr/libexec//droid-hybris/system/etc/init/`?

Reply:
Nothing reads from that folder in Android 6 or older bases. Hybris version of Android 7 base i.e. hybris-14.1 is the first one to use that folder, correct place for fixing the issue is in hybris-boot repo which seems to create the file on all Android base versions. The file is being created after this commit in hybris-boot https://github.com/mer-hybris/hybris-boot/commit/46095a4d7819962acfb49cdf88cb93d7c0a911fc

Reply:
@sledges, @mlehtima I've updated this PR as you recommended and made PR to fix original source of problem in hybris-boot (https://github.com/mer-hybris/hybris-boot/pull/184)